### ISSUE TITLE
Provide default state for form select

### ISSUE THREAD

**Issue by matthewcarleton** [patternfly/patternfly-next#1892]
Currently in the form select input we don't have a style for a "placeholder" text. It would be helpful if we styled the text to be different from the selected text option.

Here is a possible design solution.
<img width="909" alt="Screen Shot 2019-06-05 at 12 42 28 PM" src="https://user-images.githubusercontent.com/287880/58970366-b055d900-878f-11e9-8e84-ac6d58cb18cb.png">

```
font-style: italic;
color: #72767b; ($pf-color-black-600)
```
fyi: @kybaker @mceledonia

**Comment by rachael-phillips** [comment]
@kybaker @mceledonia @mcarrano what are your thoughts here? @mcoker thinks it's a fair enhancement :D

**Comment by mcarrano** [comment]
I agree with this in theory @matthewcarleton , but it does get a little bit tricky. I'm not sure that we have the idea of a default selection other than the first item in the list. We could define an option to consider the first item as a placeholder (i.e. a null selection). Otherwise the default selection needs to be an actual value (for cases where the selected value cannot be null). 

@mcoker @tlabaj would be interested in hearing your thoughts on this.

**Comment by mcoker** [comment]
I chatted with @rachael-phillips about it and she mentioned that if we do this, we should probably be consistent, and I think that implies using the same style here and in the select component. Would it make sense to make this change there, too?

Also this text looks like it might get confused as being disabled. While we have a different UI for a disabled select (grey background, no border), I still wonder if users might see the grey, italic text and wonder if it's disabled.

And just from my personal experience, it's not what I typically see in a select placeholder label, though I looked at a few other systems and material does both - grey text and italic, used in different contexts - https://material-ui.com/components/selects/

Carbon and Bootstrap display it the way we do currently

* https://www.carbondesignsystem.com/components/select/code
* https://getbootstrap.com/docs/4.0/components/forms/

**Comment by matthewcarleton** [comment]
@mcoker ya I agree this is a bit of a tricky one to tackle. On one side it looks like it could be disabled, on the other side if we leave it then it looks like it has content and might be missed. The one argument I can think of for adding these styles is even though it may appear to be disabled it won't be so it can still be tabbed it and clicked on. 
@mcarrano as far as building this we would have to test but I think we could potentially use the hidden, selected, and disabled attributes on the first option. There's a pretty good discussion around this [here](https://stackoverflow.com/questions/5805059/how-do-i-make-a-placeholder-for-a-select-box).

**Comment by andybraren** [comment]
> On one side it looks like it could be disabled, on the other side if we leave it then it looks like it has content and might be missed.

@matthewcarleton do you have the full wizard form image that prompted this issue handy? In isolation having a unique "Select X" state doesn't seem like it would matter, but in the context of a larger, dynamic form (in which pre-filled and attention-needing FormSelects could pop into the form at the same time) it was much more obvious that not having a unique "Select X" state could be an issue.

@mcoker This might be one place where FormSelect and Select could/should differ. The context this is used in seems key here. Select elements are typically isolated from others, right? FormSelect may need this unique state to differentiate from other FormSelects directly beside it in situations like the above.

> Also this text looks like it might get confused as being disabled. While we have a different UI for a disabled select (grey background, no border), I still wonder if users might see the grey, italic text and wonder if it's disabled.

Maybe just italicized and not grey? OpenShift has been using grey for "Select X", but maybe that isn't needed.

> @mcarrano as far as building this we would have to test but I think we could potentially use the hidden, selected, and disabled attributes on the first option. There's a pretty good discussion around this [here](https://stackoverflow.com/questions/5805059/how-do-i-make-a-placeholder-for-a-select-box).

+1, another [good discussion here with images](https://stackoverflow.com/questions/9447134/default-text-which-wont-be-shown-in-drop-down-list).

**Comment by mcarrano** [comment]
Ah, I was thinking this was referring to the Select component, but I keep forgetting we have both Selects and Form Selects. But wouldn't we want the same approach for these to be consistent? From a user perspective they aren't really different. Agree that the form select is perhaps a higher priority since this is where the current implementation would be most problematic.

I think this needs visual design input regarding what this might look like in the context of our overall Forms story. @mceledonia thoughts?

**Comment by matthewcarleton** [comment]
> This might be one place where FormSelect and Select could/should differ.
Can we say for sure that the select component or the dropdown component won't get used in a form or in proximity to a form? If we can't then I agree with @mcarrano these should align.
